course2/meeting4/MyLinkedList.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * MyLinkedList.h
+ *
+ *  Created on: Feb 3, 2016
+ *      Author: matthijs
+ */
+
+#ifndef MYLINKEDLIST_H_
+#define MYLINKEDLIST_H_
+
+#include <iostream>
+#include <memory>
+#include <algorithm>
+#include <iterator>
+
+template<typename T>
+struct Node
+{
+	T mValue;
+	Node* mPrev;
+	std::unique_ptr<Node> mNext;
+};
+
+template<typename T>
+class ListIterator
+{
+public:
+	//These are necessary in order to use std::copy in line 60
+	typedef T value_type;
+	typedef T* pointer;
+	typedef T& reference;
+	typedef std::ptrdiff_t difference_type;
+	typedef std::forward_iterator_tag iterator_category;
+
+	ListIterator(Node<T>* c) : mCurrent(c) { }
+
+	T& operator*() { return mCurrent->mValue; }
+	void operator++() { mCurrent = mCurrent->mNext.get(); }
+	bool operator!=(const ListIterator& o) const
+	{
+		return mCurrent != o.mCurrent;
+	}
+
+private:
+	Node<T>* mCurrent = nullptr;
+};
+
+template<typename T>
+class MyLinkedList
+{
+
+public:
+
+	typedef std::unique_ptr<Node<T>> node_ptr;
+	typedef T value_type;
+
+	MyLinkedList<T>() {}
+
+    MyLinkedList<T>(const MyLinkedList<T>& oldList) {
+    	std::copy(oldList.begin(), oldList.end(), std::front_inserter(*this));
```

Comment:
Nice, but the resulting list will have reverse order. You should use 'back_inserter', in your list implements push_back.

Usually, container have 'assign' method, that takes two iterators, and populates the content ot the range described by these iterators. If your list had this method, you could call it here.